src/shared/Core/EnvironmentBase.cs

```diff
@@ -104,7 +104,12 @@ public virtual Process CreateProcess(string path, string args, bool useShellExec
             return new Process { StartInfo = psi };
         }
 
-        public bool TryLocateExecutable(string program, out string path)
+        public virtual bool TryLocateExecutable(string program, out string path)
+        {
+            return TryLocateExecutable(program, new List<string>(), out path);
+        }
+
+        public bool TryLocateExecutable(string program, List<string> pathsToIgnore, out string path)
```

Comment:
Make this method `protected virtual` and accept an `ICollection<string>` instead

```suggestion
        protected virtual bool TryLocateExecutable(string program, ICollection<string> pathsToIgnore, out string path)
```

Reply:
The UT started complaining with `protected` so I went with `internal` instead.